apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerTestUtilsV2.java

```diff
@@ -1580,4 +1588,50 @@ static void assertExtensionCodingDoesNotExist(
 
     assertEquals(false, extensionForUrl.isPresent());
   }
+
+  /**
+   * Transform rif record to eob explanation of benefit.
+   *
+   * @param metricRegistry the {@link MetricRegistry} to use
+   * @param rifRecord the RIF record (e.g. a {@link CarrierClaim} instance) to transform@param
+   *     includeTaxNumbers whether to include tax numbers in the result (see {@link
+   *     ExplanationOfBenefitResourceProvider#HEADER_NAME_INCLUDE_TAX_NUMBERS}, defaults to <code>
+   *          false</code> )
+   * @param includeTaxNumbers if tax numbers should be included in the response
+   * @param drugCodeDisplayLookup the drug code display lookup
+   * @param npiOrgLookup the npi org lookup
+   * @return the transformed {@link ExplanationOfBenefit} for the specified RIF record
+   */
+  static ExplanationOfBenefit transformRifRecordToEob(
+      Object rifRecord,
+      MetricRegistry metricRegistry,
+      boolean includeTaxNumbers,
+      FdaDrugCodeDisplayLookup drugCodeDisplayLookup,
+      NPIOrgLookup npiOrgLookup) {
+
+    ClaimTransformerInterfaceV2 claimTransformerInterface = null;
+    if (rifRecord instanceof CarrierClaim) {
+      claimTransformerInterface =
+          new CarrierClaimTransformerV2(metricRegistry, drugCodeDisplayLookup, npiOrgLookup);
+    } else if (rifRecord instanceof DMEClaim) {
+      claimTransformerInterface = new DMEClaimTransformerV2(metricRegistry, drugCodeDisplayLookup);
+    } else if (rifRecord instanceof HHAClaim) {
+      claimTransformerInterface = new HHAClaimTransformerV2(metricRegistry, npiOrgLookup);
+    } else if (rifRecord instanceof HospiceClaim) {
+      claimTransformerInterface = new HospiceClaimTransformerV2(metricRegistry, npiOrgLookup);
+    } else if (rifRecord instanceof InpatientClaim) {
+      claimTransformerInterface = new InpatientClaimTransformerV2(metricRegistry, npiOrgLookup);
+    } else if (rifRecord instanceof OutpatientClaim) {
+      claimTransformerInterface =
+          new OutpatientClaimTransformerV2(metricRegistry, drugCodeDisplayLookup, npiOrgLookup);
+    } else if (rifRecord instanceof PartDEvent) {
+      claimTransformerInterface =
+          new PartDEventTransformerV2(metricRegistry, drugCodeDisplayLookup);
+    } else if (rifRecord instanceof SNFClaim) {
+      claimTransformerInterface = new SNFClaimTransformerV2(metricRegistry, npiOrgLookup);
+    } else {
+      throw new BadCodeMonkeyException("Unhandled RifRecord type!");
+    }
```

Comment:
This would make an awesome pattern matching switch but that's still a preview feature in java 17...